org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/preferences/Preferences.java

```diff
@@ -288,6 +329,12 @@ public static Preferences createFrom(Map<String, Object> configuration) {
 		String favoriteStaticMembers = getStringValue(configuration, FAVORITE_STATIC_MEMBERS, "");
 		prefs.setFavoriteStaticMembers(favoriteStaticMembers);
 
+		String decompilerId = getStringValue(configuration, DECOMPILER_ID_KEY, null);
```

Comment:
I'm wondering wether we should check here the decompilerId is exists, and if not, make sure the disassembler one is used instead

Reply:
I don't think so—I don't think Preferences should handle that logic, and DecompilerManager now does that internally.